src/utils/memoryCard.lua

```diff
@@ -0,0 +1,160 @@
+local pd <const> = playdate
+local ds <const> = pd.datastore
+
+class("MemoryCard").extends()
+
+local FILE_NAME <const> = "discontrolled_save_data"
+
+local DEFAULT_DATA <const> = {
+  lastPlayed = { world = 1, level = 1 },
```

Comment:
I'm seeing a small drawback in the implementation here.

`lastPlayed` should be `nil` by default, so we can detect a first game start. Then, once level 1-1 is started, we can set `lastPlayed` to level 1-1, as it's doing now.